python/cudf/cudf/core/dataframe.py

```diff
@@ -7359,8 +7359,9 @@ def select_dtypes(self, include=None, exclude=None):
                 # category handling
                 if is_categorical_dtype(i_dtype):
                     include_subtypes.add(i_dtype)
-                elif issubclass(dtype.type, i_dtype):
-                    include_subtypes.add(dtype.type)
+                elif inspect.isclass(dtype.type):
```

Comment:
`isclass` returns `True` for types like `list` and `dict` -- do we intend for those types to pass this check?

```python
>>> import inspect
>>> inspect.isclass(list)
True
>>> inspect.isclass(dict)
True
```

Reply:
yes, that's fine. whenever a list type is present in a dataframe, though (like `cudf.DataFrame({'int_col':[0,1,2],'list_col':[[1,2],[3,4],[5,6]]})`), the `dtype.type` is a `<built-in function array>`

Reply:
it doesn't look like the Struct/Series has the `select_dtypes` attribute, so it should be ok

Reply:
Sounds good - thanks!